### DISCUSSION TITLE
Cannot use API components

### DISCUSSION THREAD

**Discussion by sevyDevelopment** [quasarframework/quasar#9355] [General - Components / Directives / etc]
I am trying to use the setup function and the ref but cannot make it work, i am used to work with Vue 3 with this API Components but its not working:
![image](https://user-images.githubusercontent.com/78752710/119023647-d0c0d300-b967-11eb-86bb-dca26fa25fb9.png)

**Comment by TobyMosque** [comment]
wrap u component with `defineComponent`
https://v3.vuejs.org/api/global-api.html#definecomponent

**Reply by sevyDevelopment** [reply]
Thanks Toby for the quick answer, I did as suggested but still get an error in Vue:
![image](https://user-images.githubusercontent.com/78752710/119049493-3c199d80-b986-11eb-814f-86c3fb938844.png)
![image](https://user-images.githubusercontent.com/78752710/119049521-450a6f00-b986-11eb-87b2-73222d1480ea.png)

Whats wrong?

Thanks!

**Reply by rstoenescu** [reply]
@rafaelsevy Looking at your screenshot, I see that the error is reported by vue-router. Which can mean that you are incorrectly declaring the route. Vue Router explicitly says that the "() => import('...')" syntax is to still be used when lazy loading the component for a route. It does not matter if you wrap the actual component code with defineComponent() or not.

**Reply by sevyDevelopment** [reply]
Thank you Razvan, how can I solve this? because no matter where I use this component throws the same error, I can also see this error in my CMD.

It seeems I might be missing to install something

![image](https://user-images.githubusercontent.com/78752710/119160506-f8727280-ba1d-11eb-9e0b-db62e0ff6bf4.png)

**Reply by rstoenescu** [reply]
Show us a repo and we can help identify the misusage.

**Reply by TobyMosque** [reply]
since ref and defineComponent aren't present, i think u're using Quasar v1 + vue 2.
to use the composite APi with vue 2, u'll need to install that package:
https://www.npmjs.com/package/@vue/composition-api
